src/components/PageDefinitions.js

```diff
@@ -87,20 +88,35 @@ class PageDefinitions extends AbstractPageDefinitions {
     }
   }
 
+  // Get an array of definitions asynchronous, split them into 100 chunks and alert the user when they're all done
+  getDefinitionsAndNotify(definitions, message) {
+    const { dispatch, token } = this.props
+    const chunks = chunk(definitions, 100)
+    let todo = chunks.length
+    for (let i = 0; i < chunks.length; i++) {
+      dispatch(getDefinitionsAction(token, chunks[i])).then(() => {
+        if (--todo === 0) {
+          dispatch(uiNotificationNew({ type: 'info', message, timeout: 3000 }))
+        }
+      })
+    }
+  }
+
   refresh = () => {
-    const { components, dispatch, token } = this.props
+    const { components, dispatch } = this.props
 
     this.onRemoveAll()
     const definitions = this.buildSaveSpec(components.list)
 
-    definitions.forEach(definition => {
-      const path = definition.toPath()
+    const definitionsToGet = []
+    definitions.map(definition => {
+      definitionsToGet.push(definition.toPath())
```

Comment:
I'd like to review how refresh works from a user point of view. Ideally the data just stays there and is updated if need be. does `onRemoveAll` cause all the data to disappear? If we are really throwing away the definitions then the `delete` should not be needed.

In an ideal future state, we don't need to throw away the user's changes if there are no changes in the underlying data.